### PR TITLE
Add ipv4 forwarding instruction to kubeadm installation requirements

### PR DESCRIPTION
**Pull request by JayKayy**
Currently the `kubeadm` steps walk you through prereqs to install a cluster but walking through these steps on Centos Stream 8 i foudn that my pre-flight checks for `kubeadm init` failed with the following error:

```
[root@cp1 ~]# kubeadm init --config kubeadm-config.yml                                                                                                        
[init] Using Kubernetes version: v1.23.5                                                                                                                      
[preflight] Running pre-flight checks                                                                                                                         
error execution phase preflight: [preflight] Some fatal errors occurred:                                                                                      
        [ERROR FileContent--proc-sys-net-ipv4-ip_forward]: /proc/sys/net/ipv4/ip_forward contents are not set to 1                                            
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`                                               
To see the stack trace of this error execute with --v=5 or higher                                                                                             
```

This error can be resolved by adding an additional step for users to ensure they have already enabled ipv4 forwarding on their hosts, and provide them a simple command to do so.

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
Thank you for the PR.

This is tricky because there is already https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#letting-iptables-see-bridged-traffic

Ideally the option should be part if the same config file.

Also, this is already mentioned in https://kubernetes.io/docs/setup/production-environment/container-runtimes/ under the containerd setup. Given it's not kubeadm that needs these, it's container runtimes / containers, i am +1 to move these settings to the container runtimes page instead.

/sig cluster-lifecycle network node

**Comment by JayKayy** [comment]
Thanks for your comment! Ah, yes I didn't see it there under containerd. I had just setup a cluster using CRI-O and that step never came up. Should the similar step under containerd simply be copied to also be under the Cri-o section?

**Comment by neolit123** [comment]
I think we can:
- remove the bridge nf steps from the kubeadm page
- remove the forwarding step from containerd
- create a new section in the container runtimes page near the top that instruct how to setup bridge nf and forwarding - i.e. have it as a pre requisite for all runtimes.

**Comment by neolit123** [comment]
If you prefer to not work on this bigger change, you can help us by logging an issue. Perhaps linking to the discussion here.

**Comment by JayKayy** [comment]
No problem, I'll take it. I should be able to update the PR in a few hours

**Comment by JayKayy** [comment]
@neolit123 I'm seeing this error when following the instructions for getting hugo running locally in a container:
`▶ sudo make container-serve
[sudo] password for jkwiatko: 
podman run --rm --interactive --tty --volume /home/jkwiatko/Development/go/src/k8s.io/website:/src --cap-drop=ALL --cap-add=AUDIT_WRITE --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 -p 1313:1313 gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.87.0-b3012f7804d6 hugo server --buildFuture --environment development --bind 0.0.0.0 --destination /tmp/hugo --cleanDestinationDir
Trying to pull gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.87.0-b3012f7804d6...
Error: initializing source docker://gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.87.0-b3012f7804d6: reading manifest v0.87.0-b3012f7804d6 in gcr.io/k8s-staging-sig-docs/k8s-website-hugo: manifest unknown: Failed to fetch "v0.87.0-b3012f7804d6" from request "/v2/k8s-staging-sig-docs/k8s-website-hugo/manifests/v0.87.0-b3012f7804d6".
make: *** [Makefile:78: container-serve] Error 125
`
Any thoughts on why this pull is failing? I can't pull it outside the makefile either:

`podman pull gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.87.0-b3012f7804d6
Trying to pull gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.87.0-b3012f7804d6...
Error: initializing source docker://gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.87.0-b3012f7804d6: reading manifest v0.87.0-b3012f7804d6 in gcr.io/k8s-staging-sig-docs/k8s-website-hugo: manifest unknown: Failed to fetch "v0.87.0-b3012f7804d6" from request "/v2/k8s-staging-sig-docs/k8s-website-hugo/manifests/v0.87.0-b3012f7804d6".
`

**Comment by JayKayy** [comment]
@neolit123  Looks like the registry specified is empty? 
image

**Comment by neolit123** [comment]
Perhaps someone from the docs team can help. I haven't run hugo locally.

You can also just push commits to this PR and the CI job for netlify will generate a preview website (the CI leaves a comment on the PR with URL).

**Comment by neolit123** [comment]
https://github.com/kubernetes/website/pull/32946#issuecomment-1099810705

**Comment by JayKayy** [comment]
@neolit123 changes made. Let me know what you think.

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:59]:
```suggestion
   cat <<EOF | sudo tee /etc/modules-load.d/k8s.conf
```

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:68]:
```suggestion
   cat <<EOF | sudo tee /etc/sysctl.d/k8s.conf
```

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:71]:
```suggestion
   net.bridge.bridge-nf-call-iptables  = 1
   net.bridge.bridge-nf-call-ip6tables = 1
   net.ipv4.ip_forward                 = 1
```

slight reorder.

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (MorrisLaw) [inline content/en/docs/setup/production-environment/container-runtimes.md:78]: Should the link be this instead?

https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/#network-plugin-requirements

**Review by neolit123 — COMMENTED** [review]
this SGTM on my side, @JayKayy . thank you.

added some minor nits.

one concern would be whether the removal of the content from the "kubeadm install" page would affect students / future admins going trough https://www.cncf.io/certification/cka/ certification.

not clear if their exams have bare nodes without container runtimes (CR) and CR installation is part of the exam. possibly yes, especially after the dockershim removal in k8s 1.24.

either way, the "kubeadm install" page links to the "container runtimes" page for pre-reqs.

deferring for k8s.io docs reviews for further comments and approval of the change.

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [reply content/en/docs/setup/production-environment/container-runtimes.md:78]: k8s.io links are usually relative paths, as per the guidelines.

Comment (MorrisLaw) [reply content/en/docs/setup/production-environment/container-runtimes.md:78]: ack

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (sftim) [inline content/en/docs/setup/production-environment/container-runtimes.md:48]:
```suggestion
## {{% heading "prerequisites" %}}
```

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (sftim) [inline content/en/docs/setup/production-environment/container-runtimes.md:54]:
```suggestion
Make sure that the `br_netfilter` kernel module is loaded. This can be done by running `lsmod | grep br_netfilter`. To load it explicitly call `sudo modprobe br_netfilter`.
```

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (sftim) [inline content/en/docs/setup/production-environment/container-runtimes.md:75]: Ideally, remove the indent from these shell commands.

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (sftim) [inline content/en/docs/setup/production-environment/container-runtimes.md:50]: Do all container runtimes need this (even on Linux)? My gut feeling is they don't.

As a nice-to-have extra, explain that these steps are commonly needed but more esoteric container runtimes (I'm thinking kata containers and the like, or even a partitioning hypervisor runtime) might need a different setup.

Comment (JayKayy) [reply content/en/docs/setup/production-environment/container-runtimes.md:50]: Oh, maybe they don't _need_ this. But the kubeadm installer will [expect](https://github.com/kubernetes/kubernetes/blob/a750d8054a6cb3167f495829ce3e77ab0ccca48e/cmd/kubeadm/app/preflight/checks.go#L1030) it for a linux node regardless of the runtime.

SHould I explain that these settings are required for preflight checks in kubeadm?

Comment (neolit123) [reply content/en/docs/setup/production-environment/container-runtimes.md:50]: AFAIK Docker and containerd need the forwarding and netfilter. Overlay maybe optional in some cases. Not sure about crio overall.

These things are sparsly undocumented on kubelet and CR side.

It's not kubeadm that's needs these settings but rather the kubelet will fail foif they are not set.

So we can add a note that users can refer to the CR Foo documentation if they don't want to enable something and that these are common settings..

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:50]: re: the thread here:
https://github.com/kubernetes/website/pull/32946/files#r851613554

looks like CRIO also recommends overlay as the SD:
https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/cri-o_runtime/use-crio-engine
netfilter is also seen in external guides:
https://wiki.archlinux.org/title/CRI-O

with that i think the settings apply to * CRs that we support on this page.

suggesting the following, but language can be adjusted:

```suggestion
These instructions apply common container runtime settings for Kubernetes nodes on Linux.
If you are certain that your setup does not require a certain setting you can skip it.
Consult the individual container runtime documentation for further information.
```

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:78]: proposing adjustment accordingly:

```suggestion
```

(moved note to [top](https://github.com/kubernetes/website/pull/32946/files#r851681988))

**Review by neolit123 — COMMENTED** [review]
Thanks for the updates
/lgtm

**Review by neolit123 — COMMENTED** [review]
/lgtm
/assign @divya-mohan0209

**Comment by neolit123** [comment]
/lgtm cancel

@JayKayy 
could you please add the note in https://github.com/kubernetes/website/pull/32946#discussion_r851681988
i don't see it in the diff.

**Comment by neolit123** [comment]
@JayKayy sorry also please remove this https://github.com/kubernetes/website/pull/32946#pullrequestreview-943893714 as the networking link gets duplicated.

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:83]:
```suggestion
```

we can remove this line, since it's already mentioned here:
https://github.com/kubernetes/website/pull/32946/files#diff-e56be5b5f714d216a9a064b71a8b18655179b1f397f57a2e0dc4f76f12e46958R54

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (JayKayy) [inline content/en/docs/setup/production-environment/container-runtimes.md:215]: I did not add these lines, i need to find out how this change made it in here

Comment (neolit123) [reply content/en/docs/setup/production-environment/container-runtimes.md:214]: probably due to the rebase. you can just remove these changes from 214 to 221.

Comment (JayKayy) [reply content/en/docs/setup/production-environment/container-runtimes.md]: Done

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:83]: i still see this in the diff.

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:202]: seems like a rebase problem as well.
we already have these steps in the "Forwarding IPv4 and letting iptables see bridged traffic" section.

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:215]: we shouldn't do these changes here?

File: content/en/docs/setup/production-environment/container-runtimes.md
Comment (neolit123) [inline content/en/docs/setup/production-environment/container-runtimes.md:182]: ` Install containerd: ... Visit` is already present below.

it seems the rebase affected the diff quite a bit.

**Review by neolit123 — COMMENTED** [review]
please make the containerd section like this (exact text):

```
### containerd

Visit
[Getting started with containerd](https://github.com/containerd/containerd/blob/main/docs/getting-started.md)
and follow the instructions there, up to the point where you have a valid
configuration file, config.toml.
On Linux, you can find this file under the path `/etc/containerd/config.toml`.
On Windows, you can find this file under the path `C:\Program Files\containerd\config.toml`.

On Linux the default CRI socket for containerd is `/run/containerd/containerd.sock`.
On Windows the default CRI endpoint is `npipe://./pipe/containerd-containerd`.

#### Configuring the `systemd` cgroup driver {#containerd-systemd}

...
```

this matches the CRI-O section structure:
- install info
- cgroup driver setting as a sub-section

**Comment by neolit123** [comment]
@JayKayy this seems good to me, like this.
the sig docs approvers, might have more comments.
thanks.

/lgtm

**Review by zacharysarah — CHANGES_REQUESTED** [review]
@JayKayy Thanks for this, it's a solid improvement. ⭐ 

I've edited some bits for clarity, removed a couple of redundant headers, and put some formatting sugar on pathing in different environments.

**Comment by zacharysarah** [comment]
@JayKayy I see you already have extensive review, so I'm approving this PR to keep things moving, but I'm adding an explicit `/hold` to give you a chance to resolve feedback. When you've made the changes, feel to `/hold cancel`. 🙇🏻 

/approve
/hold

**Comment by dims** [comment]
re-apply LGTM from @neolit123 

/lgtm

**Comment by neolit123** [comment]
Perhaps someone from the docs team can help. I haven't run hugo locally.

You can also just push commits to this PR and the CI job for netlify will
generate a preview website (the CI leaves a comment on the PR with URL).

**Comment by JayKayy** [comment]
This is already merged and closed.

**Pull request merged**
